File: tests/fixtures/ecosystems/ruby_comprehensive/basic.rb
# Basic Ruby constructs for parser testing.

module Comprehensive
  VERSION = "1.0.0"

  def self.greet(name)
    "Hello, #{name}!"
  end

  class Config
    attr_accessor :env, :debug
    attr_reader :version

    def initialize(env: "development")
      @env = env
      @debug = env != "production"
      @version = VERSION
    end

    def production?
      @env == "production"
    end
  end

  class Application
    def initialize(config)
      @config = config
      @running = false
    end

    def start
      @running = true
      puts Comprehensive.greet("World")
    end

    def stop
      @running = false
    end

    def running?
      @running
    end
  end
end

File: tests/fixtures/ecosystems/python_comprehensive/basic.py
"""Basic Python constructs for parser testing."""

import os
import sys
from pathlib import Path
from typing import Optional, List

# Module-level variables
MAX_RETRIES = 3
DEFAULT_TIMEOUT = 30.0


def greet(name: str) -> str:
    """Greet a person by name."""
    return f"Hello, {name}!"


def calculate_sum(numbers: List[int]) -> int:
    """Calculate the sum of a list of numbers."""
    total = 0
    for n in numbers:
        total += n
    return total


class Config:
    """Application configuration container."""

    def __init__(self, env: str = "development"):
        self.env = env
        self.debug = env != "production"
        self.base_path = Path("/app")

    def get_setting(self, key: str, default: Optional[str] = None) -> Optional[str]:
        """Retrieve a configuration setting."""
        return os.environ.get(key, default)

    def is_production(self) -> bool:
        return self.env == "production"


class Application:
    """Main application class."""

    def __init__(self, config: Config):
        self.config = config
        self.running = False

    def start(self) -> None:
        """Start the application."""
        self.running = True
        greeting = greet("World")
        print(greeting)

    def stop(self) -> None:
        self.running = False
